src/scenes/bon_odori.inc.c

```diff
@@ -1,81 +1,356 @@
+    func_08017578();
+    D_03004b10.unk54[0xc2] = 0x7C00;
+}
+
+void func_08020880() {
+    u16 temp0;
+    u32 temp1;
+
+    func_0800c604(0);
+    temp0 = (u16) func_0800c3b8();
+    temp1 = func_08002ee0(temp0, D_089deec4[gBonOdoriInfo.unk0], 0x2000);
+    func_08005d38(temp1, func_0802085c + 1, 0);
+}
+
+void func_080208c0() {
+    u32 data;
+    func_0800c604(0);
+    data = func_080087b4(func_0800c3b8(), &D_089ded10);
+    func_08005d38(data, func_08020880, 0);
+}
+
+void func_080208ec(u32 arg0) {
+    u32 i;
+
+    gBonOdoriInfo.unk0 = arg0;
+    func_080208c0();
+    func_0800e0ec();
+    func_0800e114();
+    func_0800e0a0(0, 1, 0, -0x40, 2, 0x18, 0xC000);
+    func_0800e0a0(1, 1, 0, 0, 0, 0x1d, 1);
+    func_0800e0a0(2, 1, 0, 0, 0, 0x1e, 1);
+    gBonOdoriInfo.unk4 = func_0800c660(0x340, 2);
+    D_03004b10.unk46 = 0x311e;
+    D_03004b10.unk4C = 0x1041;
+    D_03004b10.unk4E = 0x1010;
+    
+    for (i = 0; i < 4; i++) {
+        gBonOdoriInfo.unk8[i].unk0 |= -1;
+    }
+    
+    gBonOdoriInfo.unk3C = 0;
+    gBonOdoriInfo.unk3A = 0;
+    gBonOdoriInfo.unk3E = func_0804d160(D_03005380, func_080206a0(0x1f), 0x7f, 0x78, 0x48, 0x8800, 1, 0x7f, 0);
+    gBonOdoriInfo.unk40 = 0;
+    gBonOdoriInfo.unk42 = 0;
+    func_080206c0();
+    gBonOdoriInfo.unk5C = D_089deed4[gBonOdoriInfo.unk0];
+    gBonOdoriInfo.unk58 = D_089deecc[gBonOdoriInfo.unk0];
+    func_08001ec4(0xc, 7, gBonOdoriInfo.unk5C, 0, &gBonOdoriInfo.unk460);
+    func_08001ec4(0x14, 7, gBonOdoriInfo.unk58, 0, &gBonOdoriInfo.unk60);
+    func_08017338(1, 0);
+    gBonOdoriInfo.unk864 = 0;
+}
+
+void func_08020a48() {
+    
+}
+
+
+#include "asm/scenes/bon_odori/asm_08020a4c.s" // Decomp attempt at https://decomp.me/scratch/lhu93
+
+void func_08020be4(u8 arg0) {
+    gBonOdoriInfo.unk38 = arg0;
+}
+
+void func_08020bf4(s32 arg0) {
+    func_08020a4c(arg0, gBonOdoriInfo.unk38, 0);
+}
+
+void func_08020c0c(s32 arg0) {
+    func_08020a4c(arg0, gBonOdoriInfo.unk38, 1);
+}
+
+void func_08020c24(s32 arg0) {
+    func_08020a4c(arg0, gBonOdoriInfo.unk38, 2);
+}
+
+void func_08020c3c(s32 arg0) {
+    gBonOdoriInfo.unk3C = -arg0;
+}
+
+void func_08020c4c(u32 arg0) {
+    struct BonOdoriInfo_sub* temp;
+    func_0800c604(0);
+    temp = &gBonOdoriInfo.unk8[arg0];
+    func_0804d770(D_03005380, temp->unk2, 0);
+    func_0804d8c4(D_03005380, temp->unk0, 0xa);
+}
+
+void func_08020c8c(u32 arg0) {
+    s32 temp0;
+    s32 temp1;
+    s32 temp2;
+    s16 temp3;
+    u32 temp4;
+    struct BonOdoriInfo_sub* substruct = &gBonOdoriInfo.unk8[gBonOdoriInfo.unk38];
+    if ((s16)substruct->unk0 >= 0) {
+        func_0804d770(D_03005380, substruct->unk2, 1);
+
+        temp0 = -substruct->unk8;
+        temp1 = temp0 - substruct->unkA;
+        
+        temp2 = gBonOdoriInfo.unk3C - 0x18;
+        temp2 -= gBonOdoriInfo.unk38 * 0x18;
+        D_03004b10.unkE = temp2;
+    
+        
+        func_08005d38(
+            func_0800c4b0(1, func_0800c3a4(arg0), &D_03004b10.unkC, temp0, temp1),
```

Comment:
this is confusing, would it be possible to put the result of c4b0 in a temp immediately before and then have this function on one line?